### PR TITLE
Server‑side caching Issue36

### PR DESCRIPTION
**Pull request by Boppler12**
## Summary by CodeRabbit

* **New Features**
  * Static file caching for improved performance on repeated requests.
  * URI sanitization and security checks for path handling.
  * Public cache clearing API.

* **Bug Fixes**
  * Standardized HTTP error responses for safer and consistent handling.

* **Tests**
  * Added comprehensive tests covering caching, concurrent reads, and edge cases including null-byte injection and query string handling.

### CODE REVIEW

**Review by viktorlindell12 — COMMENTED** [review]
Bra jobbat.
Koden är tydlig och lätt att följa, bra uppdelning mellan de olika delarna.

Just nu konverteras byte[] till String i build(). Detta kan bli problem senare om servern ska hantera binära filer, till exempel bilder eller PDF:er. Ett alternativ kan vara att behålla body som byte[] hela vägen och skriva ut bytes direkt till output-streamen istället.

File: src/main/java/org/example/StaticFileHandler.java [hunk 1]
```diff
@@ -12,14 +12,21 @@
 public class StaticFileHandler {
     private static final String WEB_ROOT = "www";
     private byte[] fileBytes;
+    private final FileCache cache = new FileCache();
```
[the remark this answers is not in the document]
Comment (kappsegla) [reply src/main/java/org/example/StaticFileHandler.java]: This looks like a good suggestion that the FileCache is its own concurrency safe class that gets injected to the StaticFileHandler when it is instanciated. Maybe it should be an interface implementation so StaticFileHandler just sees the interface type? That way we can implement other types of caches like using Redis.

File: src/main/java/org/example/http/HttpResponseBuilder.java [hunk 2]
```diff
@@ -1,62 +1,109 @@
...
+            Map.entry(307, "Temporary Redirect"),
+            Map.entry(308, "Permanent Redirect"),
+            Map.entry(400, "Bad Request"),
+            Map.entry(401, "Unauthorized"),
+            Map.entry(403, "Forbidden"),
+            Map.entry(404, "Not Found"),
+            Map.entry(500, "Internal Server Error"),
+            Map.entry(502, "Bad Gateway"),
+            Map.entry(503, "Service Unavailable")
+    );
```
Comment (donne41) [inline src/main/java/org/example/http/HttpResponseBuilder.java]: En PR har blivit mergad till main så en rebase för att få in det nya hit.

**Review by donne41 — CHANGES_REQUESTED** [review]
Jag känner mig lite sträng men tycker det är en del som är ändrat som inte behöver ha ändrats. Förutom det så tror jag vi börjar närma oss!
